subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/util/DefaultJavaForkOptionsTest.groovy

```diff
@@ -276,65 +276,14 @@ class DefaultJavaForkOptionsTest extends Specification {
         options.allJvmArgs == [fileEncodingProperty(), *localeProperties(), '-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005']
     }
 
-    def "debug is enabled when set using jvmArgs"() {
+    def "can set debug options"() {
         when:
-        options.jvmArgs('-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005')
```

Comment:
❓ I feel we are losing some test coverage here. Some users may already pass debug argument via the `jvmArgs` and we should keep testing for that use case.

Should we issue deprecation warning if arguments are passed via the `jvmArgs`? It may be a question too big to stall this PR for this. It's an unclear question in native as well. Basically, for modelled parameters, should users still be allowed to use the command flags or be forced to use the modelled parameters?

Reply:
If the user specifies a debug-related JVM argument, like`-Xdebg` or `-agentlib:jdwp` then Gradle now issues a warning, that the configuration in `debugOptions` will be ignored in favor of the custom arguments.